opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -545,7 +549,29 @@ def test_span_override_start_and_end_time(self):
     def test_ended_span(self):
         """"Events, attributes are not allowed after span is ended"""
 
-        with self.tracer.start_as_current_span("root") as root:
+        class ContextWrapper:
```

Comment:
@ocelotl Wow! Cool, but looks complicated :) I think this test can be simplified: we can wrap code with `assertLogs` only once and check how many logs we have at the end. Something like this:
```
    def test_ended_span(self):
        """"Events, attributes are not allowed after span is ended"""
        with self.assertLogs(level=WARNING) as _logs:
            with self.tracer.start_as_current_span("root") as root:
            ...
        self.assertEqual(len(_logs.records), 6)
```

Further if we really need we can check every log in records. But now counting logs is enough in my opinion.
What do you think?

Reply:
I say we leave it as it is :slightly_smiling_face: . It is more precise to handle each warning individually and even as it is a bit complicated right now, it serves its purpose well.

Reply:
Wouldn't it be possible to add an independent test just for it avoiding an ad-hoc class?

Reply:
Maybe, but how? :thinking: The class introduced here is just there to help catch a specific warning without affecting the catching of other warnings. I'm not sure how an independent test can help with this. It would be great if we can come up with a simpler implementation that can handle each warning independently and specifically, but I have no other ideas on how to achieve this :man_shrugging:.

Reply:
Based on @condorcet's idea, you could create a specific test case for it:

```
    def test_warning_end_span(self):
        with self.assertLogs(level=WARNING) as _logs:
            with self.tracer.start_as_current_span("root") as root:
                root.end()

        self.assertEqual(len(_logs.records), 1)
```

Reply:
I have a second thought and I think this test can be removed, it's testing if there is a warning when the context goes out, so it calls the `span.end()`, we already have a test to check if the context calls `span.end()`. The test if `span.end()` prints a warning is already present on this PR.

Reply:
Another way is just using `start_span` manually instead of wrapping code by `start_as_current_span` context manager that spam with warning when try to `end()` span. For testing purpose It will be enough.
Something like this:
```
    def test_ended_span(self):
        """"Events, attributes are not allowed after span is ended"""

        root = self.tracer.start_span("root")
        # everything should be empty at the beginning
        self.assertEqual(len(root.attributes), 0)
        ...
```

Reply:
Ok, refactored, please let me know what you think now @condorcet, @mauriciovasquezbernal. :+1: